src/cdk/schema/cdk.json

```diff
@@ -0,0 +1,233 @@
+{
```

Comment:
The correct long term place for this file should probably be in the CDK repo and maintained by the CDK team.

There is no forcing function for us to maintain it, nor are we in the loop when it gets out of date. This would also allow us to share the file with the JetBrains and VS IDEs as well.

Have you tried to submit this to them?

Reply:
This is misunderstanding of what `cdk.json` is for CDK (what I referred above as _mystified_).  CDK _per se_ doesn't depend on `cdk.json` (we can have completely correct CDK project without any _cdk.json_ - just need to specify `app` from command line). 
So, `cdk.json` is all about storing default values in a specific place. You can also store any arbitrary values there, just for your own pleasure or for a company extension. 
And this VSCode extension is about making work experience with CDK better, so, this JSON schema helps on that. 
CDK is pretty old project for now and if they didn't made the schema yet that means they don't care about editing/validating experience and left that to a third-party extensions. That's why I'm here. 

Do you think this additional will be useful for users of this extensions? I'm 100% sure in that.

Reply:
@tinovyatkin thank you for this! It seems useful for users beyond VSCode. 

Before rushing to merge this PR, we would like to see if we can find an upstream home for this schema. I left a comment in https://github.com/aws/aws-cdk/issues/12799

Reply:
Thank you @justinmk3, your answer sounds like from a _customer obsessed_ company employee. I'll be happy to land the JSON schema in a different place, but ultimate contribution point for VSCode users (and so, for the unit test) will be this extension. So, we may considering merge in the meanwhile (to make customers happy) and if an upstream owner arise then just to change reference to the schema.

Reply:
To have the most impact imho, the best place to put this schema would be to update the existing one in SchemaStore

https://www.schemastore.org/json/

https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/cdk.json

https://code.visualstudio.com/docs/languages/json#_json-schemas-and-settings
https://www.jetbrains.com/help/idea/json.html#ws_json_using_schemas

Reply:
@arnulfojr thanks for mentioning that! Your comment would be helpful at https://github.com/aws/aws-cdk/issues/12799 also :)

We will look at leveraging schemastore as source of truth for AWS Toolkit (or does vscode do it automatically?)